### ISSUE TITLE
tuneScheme() function should be in miscr.R but is not

### ISSUE THREAD

**Issue by noeliarico** [topepo/caret#1003]
Hi everyone,

I'm developing an experimental classification method and I would be interested in not fitting the model for all the rows of the grid parameter. I want to keep all its possible combinations though. 

This is because the real stuff happens in the prediction. Just imagine we were talking about knn with a grid with different values for the k, the distances from the instances of the testing set to the instances of the training set would be the same for all the possible values of k, only the output would change based on the number of instances being considered. Well, my method is not knn and the equivalent step of calculating distances is too time consuming, so it would be a great advantage avoid it when possible.

In the comments of the`train.default` function in https://github.com/topepo/caret/blob/master/pkg/caret/R/train.default.R it says that:

    ## trainInfo will hold the information about how we should loop to train the model and what types
    ## of parameters are used.

    ## There are two types of methods to build the models: "basic" means that each tuning parameter
    ## combination requires it's own model fit and "seq" where a single model fit can be used to
    ## get predictions for multiple tuning parameters.

    ## The tuneScheme() function is in miscr.R and it helps define the following:
    ##   - A data frame called "loop" with columns for parameters and a row for each model to be fit.
    ##     For "basic" models, this is the same as the tuning grid. For "seq" models, it is only
    ##     the subset of parameters that need to be fit
    ##   - A list called "submodels". If "basic", it is NULL. For "seq" models, it is a list. Each list
    ##     item is a data frame of the parameters that need to be varied for the corresponding row of
    ##     the loop object.

But I'm not able to find this function anywhere, nor an explanation about where I should use it. Does anyone knows how could I solve this?

Thank you in advance :)

Noelia

**Comment by topepo** [comment]
That message needs to be updated although the tactics are still valid. 

Each code module (e.g. `glm`, `rf`, and others found [here](https://github.com/topepo/caret/tree/master/models/files)) has an optional slot called `loop` that is what the documentation called `tuneScheme`. It creates the two components mentioned in those comments. 

Better documentation and an example are found [here](https://topepo.github.io/caret/using-your-own-model-in-train.html#illustrative-example-2-something-more-complicated---logitboost). 

I'll update these comments.
